Modules/Loadable/Markups/VTKWidgets/vtkSlicerMarkupsWidgetRepresentation2D.cxx

```diff
@@ -288,13 +400,163 @@ void vtkSlicerMarkupsWidgetRepresentation2D::UpdateAllPointsAndLabelsFromMRML(do
         continue;
         }
 
+      // get control point position
       double slicePos[3] = { 0.0 };
       this->GetNthControlPointDisplayPosition(pointIndex, slicePos);
-
       controlPoints->ControlPoints->InsertNextPoint(slicePos);
-      slicePos[0] += labelsOffset / sqrt(2.0);
-      slicePos[1] += labelsOffset / sqrt(2.0);
-      this->Renderer->SetDisplayPoint(slicePos);
+
+      // set normal label position
+      double labelPos[3] = { 0.0 };
+      labelPos[0] = slicePos[0] + labelsOffset / sqrt(2.0);
+      labelPos[1] = slicePos[1] + labelsOffset / sqrt(2.0);
+
+      std::string labelStr;
+      int labelLength = 0;
+
+      labelStr = markupsNode->GetNthControlPointLabel(pointIndex);
+
+      // make label texts multiline if <br> is used
+      std::string from = "<br>";
```

Comment:
This would be a very expensive operation when there are thousands of points. Do we really need this? It would be simpler and more efficient if we we made sure that the newline character (`\n`) can be properly entered on the GUI. This could be done in a separate pull request.